### PR TITLE
Adding the user.sls file for user creation using salt

### PR DESCRIPTION
**Pull request by annweshasarma**
## Here are some reminders before you submit the pull request
- [ ] Add tests for the change
- [ ] Document changes
- [ ] Communicate in the mailing list if needed
- [ ] Pass `make installcheck`
- [ ] Review a PR in return to support the community

### CODE REVIEW

File: gpMgmt/bin/go-tools/configs/salt/common/user.sls [hunk 1]
```diff
@@ -0,0 +1,16 @@
+add_gpadmin_group:
+  group.present:
+    - name: gpadmin
```
Comment (ajs95) [inline gpMgmt/bin/go-tools/configs/salt/common/user.sls:3]: variable values can be added as a pillar file

File: gpMgmt/bin/go-tools/configs/salt/common/user.sls [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+
+create_gpadmin_user:
+  user.present:
+    - name: gpadmin
+    - createhome: True
+    - home: /home/gpadmin
+    - shell: /bin/bash
+    - password: changeme1!
```
Comment (ajs95) [inline gpMgmt/bin/go-tools/configs/salt/common/user.sls:11]: should be in pillar file

File: gpMgmt/bin/go-tools/configs/salt/common/user.sls (hunk 1 above)
Comment (ajs95) [reply gpMgmt/bin/go-tools/configs/salt/common/user.sls:3]: we can change the directory structure as

```
configs/
|_ saltconfigs/
    |__  salt/
    |__  pillars/
```

File: gpMgmt/bin/go-tools/configs/salt/common/user.sls [hunk 3]
```diff
@@ -0,0 +1,16 @@
+add_gpadmin_group:
+  group.present:
+    - name: gpadmin
+
+create_gpadmin_user:
+  user.present:
+    - name: gpadmin
```
Comment (ajs95) [inline gpMgmt/bin/go-tools/configs/salt/common/user.sls:7]: reuse variable from pillar file

File: gpMgmt/bin/go-tools/configs/salt/common/user.sls (hunk 1 above)
Comment (annweshasarma) [reply gpMgmt/bin/go-tools/configs/salt/common/user.sls]: Used a pillar file for user details and creation and changed the directory name

File: gpMgmt/bin/go-tools/configs/salt/common/user.sls (hunk 2 above)
Comment (annweshasarma) [reply gpMgmt/bin/go-tools/configs/salt/common/user.sls]: Added a pillar file

**Pull request merged**